cmd/gossamer/README.md

```diff
@@ -0,0 +1,190 @@
+The flags/parameters that the Gossamer CLI supports are defined in [`flags.go`](flags.go). For an exhaustive reference
+of the Gossamer CLI capabilities, follow [the installation instructions](../../README.md#installation) and execute
+`./bin/gossamer --help`.
+
+### Default Command
+
+This the default Gossamer execution method, which invokes the `gossamerAction` function defined in
+[`main.go`](main.go) - it will launch a Gossamer blockchain client. The details of how Gossamer orchestrates a
+blockchain client are [described below in the Client Components section](#client-components).
+
+- `basepath` - the path to the directory where Gossamer will store its data
+- `chain` - specifies the [chain configuration](../../chain) that the
+  [Gossamer host node](https://chainsafe.github.io/gossamer/getting-started/overview/host-architecture/) should load
+- `key` - specifies a test keyring account to use (e.g. `--key=alice`)
+- `log` - supports levels `crit` (silent), `error`, `warn`, `info`, `debug`, and `trce` (detailed), default is `info`
+- `name` - node name, as it will appear in, e.g., [telemetry](https://telemetry.polkadot.io/)
+
+### Init Subcommand
+
+This subcommand accepts a genesis configuration file and uses it to initialise the Gossamer node and its state. The
+`init` subcommand invokes the `initAction` function defined in [`main.go`](main.go).
+
+- `genesis` - path to the "compiled" genesis configuration file that should be used to initialise the Gossamer node and
+  its state
+
+### Account Subcommand
+
+The `account` subcommand provides the user with capabilities related to generating and using `ed25519`, `secp256k1`, and
+`sr25519` [account keys](https://wiki.polkadot.network/docs/learn-keys), and managing the keys present in the
+[Gossamer keystore](#keystore). The `actionAction` function is defined in [account.go](account.go); it is an interface
+to the capabilities defined in the [`lib/crypto`](../../lib/crypto) and [`lib/keystore`](../../lib/keystore) packages.
+This subcommand provides capabilities that are similar to
+[Parity's Subkey utility](https://docs.substrate.io/v3/tools/subkey).
+
+- `generate` - creates a new key pair; specify `--ed25519`, `--secp256k1`, or `--sr25519` (default)
+- `list` - lists the keys in the Gossamer keystore
+- `password` - allows the user to provide a password to either encrypt a generated key or unlock the Gossamer keystore
+
+### Import Runtime Subcommand
+
+This subcommand takes a [Wasm runtime binary](https://wiki.polkadot.network/docs/learn-wasm) and uses it to generate a
+[genesis](https://wiki.polkadot.network/docs/glossary#genesis) configuration file; it does not require any flags, but
+expects the path to a Wasm file to be provided as a command-line parameter (example:
+`./bin/gossamer import-runtime runtime.wasm > genesis.json`). The `import-runtime` subcommand invokes the
+`importRuntimeAction` function defined in [`main.go`](main.go).
```

Comment:
I'm thinking a summary markdown table with columns subcommand, main.go function and link to subsection here, at the top would be nice. Just a suggestion though.

Reply:
Definitely not a bad idea, but I'm going to put a pin in this for now.